### PR TITLE
Rolling update

### PR DESCRIPTION
**Pull request by t27gupta**
- need readiness check only
- create a new section "Rolling update" in README before "Scaling a deployment" section to explain RollingUpdate
  - update the yaml file only
  - add the RollingUpdate and readinessProbe configuration
    - readinessProbe should be as simplest as possible
  - tell that readiness check class is provided
  - direct users to visit kube-mp-health guide for other health checks

### CODE REVIEW

**Review by gkwan-ibm — CHANGES_REQUESTED** [review]
License year should be `2022`

**Review by gkwan-ibm — CHANGES_REQUESTED** [review]
@t27gupta 
- fix the test failure
- add hotspot to rollingUpdate -> "The `rollingUpdate` configuration..."
  - highlight all following lines for both services
```
  -   strategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 1
      maxSurge: 1
```
- "...has two attributes within it which are `maxUnavailable` and `maxSurge`. " ->  "...has two attributes. "
- "maxUnavailable refers ..." -> "The `maxUnavailable` attribute refers ..."
  - hotspot `maxUnavailable`
- "...the the number ", remove a "the"
- "Similarly, maxSurge refers..." -> "Similarly, the `maxSurge` attribute refers..."
  - hotspot `maxSurge`
- rewrite "The readiness check for both Inventory and System has already been provided within the health directory of each."
   - a statement to explain why need the readiness check
   - a statement tells the readiness check for both services are provided
   - hotspot should highlight all following lines
```
        readinessProbe:
          httpGet:
            path: /health/ready
            port: 9080
          initialDelaySeconds: 30
          periodSeconds: 10
          timeoutSeconds: 3
          failureThreshold: 1
```
- "The remaining startup and liveness health checks can be found in the [Kubernetes-microprofile-health](https://openliberty.io/guides/kubernetes-microprofile-health.html) guide." -> "You can learn more about how to use health checks in Kubernetes, check out the [Kubernetes-microprofile-health](https://openliberty.io/guides/kubernetes-microprofile-health.html) guide."

**Comment by gkwan-ibm** [comment]
Readiness checklist:
- [x] verify on the guides development site VM
- [x] verify end-to-end locally if the source has change
- [x] verify end-to-end on cloud-hosted if apply
- [x] PR test pass

**Comment by gkwan-ibm** [comment]
hi @t27gupta 

- add hotspot to the rollingUpdate word -> "The `rollingUpdate` configuration..."
  - highlight all following lines for both services
```
  -   strategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 1
      maxSurge: 1
```
- "...attributes maxUnavailable and maxSurge. " backtick both attributes as "...attributes `maxUnavailable` and `maxSurge`."
- the `readiness check` hotspot no need to highlight following lines
```
        env:
        - name: SYS_APP_HOSTNAME
          value: system-service
```

**Comment by gkwan-ibm** [comment]
hi @t27gupta
- "Rolling Update refers to... " -> "Rolling update refers to ..."
- "Update the yaml as follows to add the rollingUpdate configuration." -> "Update the yaml file as follows to add the `rollingUpdate` configuration."
    - add "file" after yaml and backtick "rollingUpdate"
- "The `readiness check` allows third party applications and services to know whether the service is ready to handle requests. " -> "The `readinessProbe` configuration allows Kubernetes to know whether the service is ready to handle requests." 
  -  `readinessProbe` highlights same as `readiness check`
- "A readiness health check has already been implemented for the inventory and system services." -> "The readiness health check classes for the `/health/ready` endpoint to the `inventory` and `system` services are provided for you."
- "You can learn more about..." -> "If you want to learn more about..."

File: README.adoc [hunk 1]
```diff
@@ -356,6 +356,26 @@ endif::[]
 // Scaling a deployment
 // ================================================================================================
 
+== Rolling update
+
+Rolling update refers to continually installing Kubernetes patches without disrupting the availability of the deployed applications. Without continuous updates, a Kubernetes cluster will be susceptible to a denial of a service attack. Update the yaml file as follows to add the `rollingUpdate` configuration.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
Without continuous updates, a Kubernetes cluster is susceptible to a denial of a service attack. Rolling updates continually install Kubernetes patches without disrupting the availability of the deployed applications. Update the yaml file as follows to add the `rollingUpdate` configuration. 
```

File: README.adoc [hunk 2]
```diff
@@ -356,6 +356,26 @@ endif::[]
...
+`kubernetes.yaml`
+----
+
+kubernetes.yaml
+[source, yaml, linenums, role="code_column"]
+----
+include::finish/kubernetes.yaml[]
+----
+
+The [hotspot=rolling1 hotspot=rolling2]`rollingUpdate` configuration has two attributes `maxUnavailable` and `maxSurge`. The [hotspot=maxUnavailable1 hotspot=maxUnavailable2]`maxUnavailable` attribute refers to the the number of Kubernetes pods that are allowed to be unavailable during the update process. Similarly, the [hotspot=maxSurge1 hotspot=maxSurge2]`maxSurge` attribute refers to the number of additional pods that are allowed to be created during the update process.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The [hotspot=rolling1 hotspot=rolling2]`rollingUpdate` configuration has two attributes, `maxUnavailable` and `maxSurge`. The [hotspot=maxUnavailable1 hotspot=maxUnavailable2]`maxUnavailable` attribute specifies the the maximum number of Kubernetes pods that can be unavailable during the update process. Similarly, the [hotspot=maxSurge1 hotspot=maxSurge2]`maxSurge` attribute specifies the maximum number of additional pods that can be created during the update process.
```

**Review by dmuelle — COMMENTED** [review]
ID review- a few minor suggestions

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Review by gkwan-ibm — APPROVED** [review]

**Pull request merged**